pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -179,6 +186,37 @@ func (r *Reconciler) getOrCreateClusterPolicySnapshot(ctx context.Context, crp *
 	return latestPolicySnapshot, nil
 }
 
+func (r *Reconciler) deleteRedundantSchedulingPolicySnapshots(ctx context.Context, crp *fleetv1beta1.ClusterResourcePlacement, sortedList *fleetv1beta1.ClusterSchedulingPolicySnapshotList) error {
+	var err error
+	if sortedList == nil {
```

Comment:
This will happen every time if the function earlier does not return a sortedList?